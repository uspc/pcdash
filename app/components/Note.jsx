import React from 'react';
import {DragSource,DropTarget} from 'react-dnd';
import ItemTypes from '../constants/itemTypes';
import {compose} from 'redux';



const Note = ({
    connectDragSource, connectDropTarget,isDragging,isOver,onMove,id, children, ...props
    }) => {
  return compose(connectDragSource,connectDropTarget)(
      <div style={{opacity: isDragging || isOver ? 0 : 1}} {...props}>
        {children}
      </div>
  );
};

const noteSource = {
  beginDrag(props) {
    console.log('begin dragging note', props);
    return {
        id: props.id
    };
  }
};

const noteTarget = {
    hover(targetProps,monitor){
        const sourceProps = monitor.getItem();
        const targetId = targetProps.id;
        const sourceId = sourceProps.id;

        if(sourceId !== targetId){
            targetProps.onMove({sourceId, targetId})
        }
    }
};

export default compose(
    DragSource(ItemTypes.NOTE,noteSource,(connect,monitor) => ({
        connectDragSource: connect.dragSource(),
        isDragging: monitor.isDragging()
    })),
    DropTarget(ItemTypes.NOTE,noteTarget,(connect,monitor) => ({
        connectDropTarget: connect.dropTarget(),
        isOver: monitor.isOver()
    }))
)(Note)