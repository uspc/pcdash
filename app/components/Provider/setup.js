import storage from '../../libs/storage';
import persist from '../../libs/persist';
import NoteStore from '../../stores/NoteStore';
import LaneStore from '../../stores/LaneStore';
import loadseed from '../../libs/loadseed';

export default alt => {
  alt.addStore('NoteStore', NoteStore);
  alt.addStore('LaneStore', LaneStore);

  loadseed(sessionStorage,'app',alt);
  //persist(alt, storage(sessionStorage), 'app');
}


