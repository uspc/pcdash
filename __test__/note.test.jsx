import renderer from 'react-test-renderer';
import React from 'react';
import Note from '../app/components/Note';
import Provider from '../app/components/Provider';
import {DragDropContext} from 'react-dnd';
import {renderIntoDocument} from 'react-addons-test-utils';
import TestBackend from 'react-dnd-test-backend';
import assert from 'assert';

test('Testing Note', () => {

    const test = 'test';
    const NoteContent = wrapInTestContext(Note);
    const tree = renderer.create(<NoteContent id="demo" isDragging="0" isOver="0">{test}</NoteContent>).toJSON();
    expect(tree).toMatchSnapshot();
    //expect(tree.children).toEqual([test]);
});

// Option 1 wrapping using decorator pattern
function wrapInTestContext(DecoratedComponent) {
    @DragDropContext(TestBackend)
    class TestContextContainer extends React.Component {
        render() {
            return <DecoratedComponent {...this.props} />;
        }
    }
    return TestContextContainer;
}

// Option 2 wrapping using return type
function wrapTestContext(DecoratedComponent){
    return DragDropContext (TestBackend)(
        React.createClass({
            render(){
                return <DecoratedComponent {...this.props} />;
            }
        })
    );
}