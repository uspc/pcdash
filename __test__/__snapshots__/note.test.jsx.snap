exports[`test Testing Note 1`] = `
<div
  style={
    Object {
      "opacity": 1,
    }
  }>
  test
</div>
`;
